### PR TITLE
Add RegExp query doc

### PR DESCRIPTION
**Pull request by steelywing**

### CODE REVIEW

File: pages/queries.rst [hunk 1]
```diff
@@ -45,6 +45,10 @@ Messages that do not have the field *type*::
 
   NOT _exists_:type
 
+Messages that match RegExp *FastEthernet[0-9]+* (ElasticSearch will convert messages to lowercase)::
```
Comment (joschi) [inline pages/queries.rst]: Please write "regular expression" instead of "RegExp".

Comment (joschi) [inline pages/queries.rst]: "ElasticSearch" -> "Elasticsearch"

Comment (joschi) [inline pages/queries.rst]: In fact, Elasticsearch will only convert the query to lower case, if the configured tokenizer does that:
https://www.elastic.co/guide/en/elasticsearch/reference/5.6/analysis-lowercase-tokenizer.html

For example, the standard analyzer of Elasticsearch is using the lowercase tokenizer:
https://www.elastic.co/guide/en/elasticsearch/reference/5.6/analysis-standard-analyzer.html

File: pages/queries.rst [hunk 2]
```diff
@@ -45,6 +45,10 @@ Messages that do not have the field *type*::
...
+
+  /fastethernet[0-9]+/
```
Comment (joschi) [inline pages/queries.rst]: Either change to "/FasterEthernet[0-9]+/" here or to "fasterethernet[0-9]+" in the above paragraph.

File: pages/queries.rst (hunk 1 above)
Comment (joschi) [inline pages/queries.rst]: Please use back-quotes to emphasize the regular expression:
http://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#inline-markup

```text
``FastEthernet[0-9]+``
```

**Review by joschi — CHANGES_REQUESTED** [review]

File: pages/queries.rst (hunk 1 above)
Comment (steelywing) [reply pages/queries.rst]: But if I search `FastEthernet[0-9]+`, the result is empty, if I search `fastethernet[0-9]+`, the result is normal, may be I just use all lowercase in this example?

Comment (joschi) [reply pages/queries.rst]: As explained, it depends on the tokenizer which is being used for the respective field(s).

Maybe add examples and an explanation for when lowercase searches work and when not.

Comment (steelywing) [reply pages/queries.rst]: Actually I don't know what is tokenizer? 😄 I just know the lowercase works...

Comment (steelywing) [reply pages/queries.rst]: Can you help to add the explaination? 😃 @joschi

**Review by joschi — DISMISSED** [review]

**Pull request merged**